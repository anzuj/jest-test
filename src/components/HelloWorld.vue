<template>
  <v-container>
    <h4>{{ title }}</h4>
    <v-row align="center">
      <v-col cols="3">
        <div id="nr" class="mr-6">{{ count }}</div>
      </v-col>
      <v-col class="d-flex">
        <v-btn
          id="incrementBtn"
          class="mx-2"
          small
          outlined
          color="indigo"
          @click="increment($event)"
          ><v-icon>mdi-plus</v-icon>
        </v-btn>
        <br />
        <!-- <v-btn id="resetBtn" small outlined color="grey" @click="$emit('reset')"
          >Reset</v-btn
        > -->
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
export default {
  props: ["title", "count"],
  data: () => ({}),
  methods: {
    increment() {
      this.$emit("increment");
    },
  },
};
</script>

<style>
#nr {
  font-size: 50px;
  font-weight: bold;
}
</style>
