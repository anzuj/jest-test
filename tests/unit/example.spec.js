// Libraries
// import Vue from "vue";
import Vuetify from "vuetify";

// Components
import HelloWorld from "@/components/HelloWorld.vue";

// Utilities
import { createLocalVue, mount } from "@vue/test-utils";

// OPTIONAL HELPER MOUNT FUNCTION
const mountFunction = (options) => {
  return mount(HelloWorld, {
    localVue,
    vuetify,
    ...options,
  });

  // can be used as:
  // it('should have a custom title and match snapshot', () => {
  //   const wrapper = mountFunction({
  //     propsData: { title: 'Fizzbuzz' },
  //   })
};

describe("HelloWorld.vue", () => {
  const localVue = createLocalVue();
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  // TESTS BELOW

  // --------------------------------------------------------------------------------------

  it("Mounts properly", () => {
    const wrapper = mount(HelloWorld, {
      localVue,
      vuetify,
    });

    expect(wrapper.vm).toBeTruthy();
  });

  // --------------------------------------------------------------------------------------

  it("Title prop displays correctly", () => {
    const wrapper = mount(HelloWorld, {
      localVue,
      vuetify,
      propsData: { title: "Foobar" },
    });

    //option 1: we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();

    // Option 2a: we can locate relevant part in html (css selectors)
    // const title = wrapper.find('h4')

    // 2b and expect it to match the given prop
    // expect(title.text()).toBe('Foobar')
  });
  // --------------------------------------------------------------------------------------
});

// 2 -  EMIT PARENT EVENT
/*   it('should emit an event when the action v-btn is clicked', () => {
    const wrapper = mount(CustomCard, {
      localVue,
      vuetify,
      propsData: { title: 'Foobar' },
    })

    const event = jest.fn()
    const button = wrapper.find('.v-btn')

    // Here we bind a listener to the wrapper
    // instance to catch our custom event
    // https://vuejs.org/v2/api/#Instance-Methods-Events
    wrapper.vm.$on('action-btn:clicked', event)

    expect(event).toHaveBeenCalledTimes(0)

    // Simulate a click on the button
    button.trigger('click')

    // Ensure that our mock event was called
    expect(event).toHaveBeenCalledTimes(1)
  }) */

// 3 - just VUE
// describe('HelloWorld.vue', () => {
//   it('renders props.title when passed', () => {
//     const title = 'new message'
//     const wrapper = shallowMount(HelloWorld, {
//       propsData: { title }
//     })
//     expect(wrapper.text()).toMatch(title)
//   })
// })

// 4 - VUETIFY IF MOUNTS
/*
import Vuetify from 'vuetify'
import { mount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

const vuetify = new Vuetify()
const wrapper = mount(HelloWorld , {
  vuetify
})

describe('HelloWorld .vue:', () => {
  it('1. Mounts properly', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })
}) 
*/



