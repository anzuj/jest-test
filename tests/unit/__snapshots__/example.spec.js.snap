// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue Title prop displays correctly 1`] = `
<div class="container">
  <div class="row text-center">
    <div class="col col-12">
      <h4>Foobar</h4>
      <div class="v-image v-responsive my-3 theme--light" style="height: 100px;">
        <div class="v-responsive__content"></div>
      </div>
    </div>
  </div>
</div>
`;
